jsesc.js

```diff
@@ -241,44 +241,16 @@ const jsesc = (argument, options) => {
 	result = '';
 	while (++index < length) {
 		const character = string.charAt(index);
-		if (options.es6) {
-			const first = string.charCodeAt(index);
-			if ( // check if it’s the start of a surrogate pair
-				first >= 0xD800 && first <= 0xDBFF && // high surrogate
-				length > index + 1 // there is a next code unit
-			) {
-				const second = string.charCodeAt(index + 1);
-				if (second >= 0xDC00 && second <= 0xDFFF) { // low surrogate
-					// https://mathiasbynens.be/notes/javascript-encoding#surrogate-formulae
-					const codePoint = (first - 0xD800) * 0x400 + second - 0xDC00 + 0x10000;
-					let hexadecimal = codePoint.toString(16);
-					if (!lowercaseHex) {
-						hexadecimal = hexadecimal.toUpperCase();
-					}
-					result += '\\u{' + hexadecimal + '}';
-					++index;
-					continue;
-				}
-			}
-		}
 		if (!options.escapeEverything) {
 			if (regexWhitelist.test(character)) {
 				// It’s a printable ASCII character that is not `"`, `'` or `\`,
 				// so don’t escape it.
 				result += character;
 				continue;
 			}
-			if (character == '"') {
-				result += quote == character ? '\\"' : character;
-				continue;
-			}
-			if (character == '`') {
-				result += quote == character ? '\\`' : character;
-				continue;
-			}
-			if (character == '\'') {
-				result += quote == character ? '\\\'' : character;
-				continue;
+			if (character == '"' || character == '`' || character == '\'') {
```

Comment:
Wouldn't this be better?

```js
if (character == quote) {
  result += '\\' + quote;
  continue;
}
```

Reply:
That was my original code, but if it's a quote char and not equal to `quote`, we return it without escaping.